core/src/software/aws/toolkits/core/utils/NullabilityUtils.kt

```diff
@@ -0,0 +1,14 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.core.utils
+
+/**
+ * Invokes the [block] if the receiving type is null
+ */
+fun <T : Any?> T.onNull(block: () -> Unit): T {
```

Comment:
The opposite of `let`.... hmmm

I like.

`?:` doesn't let you do a code block

Reply:
`?:` still evaluates this to determine the overall statement result so you get `T|Unit` or something weird - I wanted to maintain the fact it's `null` but do something in that case.

Reply:
Ya, I like the addition.